samples/07.advanced-web-chat-apps/e.sso-on-behalf-of-authentication/README.md

```diff
@@ -0,0 +1,250 @@
+   -  Vice versa, end-user is able to sign in through the bot, and is recognized by the web page immediately
+-  End-user is able to sign in through the web page and sign out though the bot
+   -  Vice versa, end-user is able to sign in through the bot and sign out through the web page
+
+## Content of the `.env` files
+
+The `.env` files hold the environment variables critical to run the service. These are usually security-sensitive information and must not be committed to version control. Although we recommend keeping these keys in Azure Vault, for simplicity of this sample, we would keep them in `.env` files.
+
+To ease the setup of this sample, here is the template of `.env` files.
+
+### `/bot/.env`
+
+```
+MicrosoftAppId=12345678-1234-5678-abcd-12345678abcd
+MicrosoftAppPassword=a1b2c3d4e5f6
+connectionName=botOAuthConnection
+```
+
+### `/rest-api/.env`
+
+```
+AAD_OAUTH_CLIENT_ID=12345678abcd-1234-5678-abcd-12345678abcd
+AAD_OAUTH_REDIRECT_URI=http://localhost:3000/api/aad/oauth/callback
+AAD_OAUTH_TENANT_ID=abcd1234-abcd-1234-efgh-5678abcdefgh
+DIRECT_LINE_SECRET=a1b2c3.d4e5f6g7h8i9j0
+```
+
+## OAuth provider support single redirect URI only
+
+In this sample, we do not use OAuth card due to technical limitations on some OAuth providers which support single redirect URI only.
+
+In order to use the website to sign in, the developer will need to set the redirect URI to their own web API.
+
+In order to use the bot to sign in, in the OAuth provider, the developer will need to set the redirect URI to https://token.botframework.com/.auth/web/redirect.
+
+Since some OAuth providers do not support multiple redirect URIs, we prefer using a single redirect URI from the web API to make sure existing authorization flow is not disturbed.
+
+# Frequently asked questions
+
+## How can I reset my authorization?
+
+After having signed in on this app, click the profile photo on the upper-right hand corner, select "Review access on Office.com". Then, you will be redirected to the OAuth provider page to remove your authorization.
+
+1. On the AAD dashboard page, wait until "App permissions" loads. Here you see how many apps you have authorized
+1. Click "Change app permissions"
+1. In the "You can revoke permission for these apps" section, click the "Revoke" button below your app registration
+
+# Further reading
+
+## Related articles
+
+-  [RFC 6749: The OAuth 2.0 Authorization Framework](https://tools.ietf.org/html/rfc6749)
+-  [RFC 6819: OAuth 2.0 Threat Model and Security Considerations](https://tools.ietf.org/html/rfc6819)
+-  [RFC 7636: Proof Key for Code Exchange by OAuth Public Clients](https://tools.ietf.org/html/rfc7636)
+-  [IETF Draft: OAuth 2.0 for Browser-Based Apps](https://tools.ietf.org/html/draft-ietf-oauth-browser-based-apps-01)
+-  [Bot Framework Blog: Enhanced Direct Line Authentication feature](https://blog.botframework.com/2018/09/25/enhanced-direct-line-authentication-features/)
+
```

Comment:
Should we add this article?

https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow